src/sdam/server.ts

```diff
@@ -310,7 +310,11 @@ export class Server extends TypedEventEmitter<ServerEvents> {
     this.incrementOperationCount();
     if (conn == null) {
       try {
-        conn = await this.pool.checkOut();
+        if (options.operationTimeout) {
+          conn = await this.pool.checkOut({ timeout: options.operationTimeout });
+        } else {
+          conn = await this.pool.checkOut();
+        }
```

Comment:
I feel Warren's current code is easier to read, but if we do end up going with this suggestion can we leave a clarifying comment?

Reply:
I am surprised because breaking this up into two calls to checkOut based on a condition that does matter is more to read without more meaningful context given. Whether or not timeout exists, there is no change to how checkOut is, practically, invoked because the typescript reports that field as optional.

I would actually take this further:
```js
conn = await this.pool.checkOut(options);
```
Why do we need to make a new object here? passing through options should be fine right? Less branching paths the less there is to debug

Reply:
Can you elaborate on what may accidentally break the spec without a test warning us?